sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClientTest.java

```diff
@@ -113,8 +123,15 @@ void setup() {
         connectionProcessor = Mono.fromCallable(() -> connection).repeat(10).subscribeWith(
             new ServiceBusConnectionProcessor(connectionOptions.getFullyQualifiedNamespace(),
                 connectionOptions.getRetry()));
-        sender = new ServiceBusSenderAsyncClient(ENTITY_NAME, connectionProcessor, retryOptions,
-            tracerProvider, messageSerializer, onClientClose);
+
+        sender = new ServiceBusSenderAsyncClient(ENTITY_NAME, MessagingEntityType.QUEUE, connectionProcessor,
+            retryOptions, tracerProvider, messageSerializer, onClientClose);
+
+        when(connection.getManagementNode(anyString(), any(MessagingEntityType.class)))
+            .thenReturn(just(managementNode));
+
+        when(sendLink.getLinkSize()).thenReturn(Mono.just(ServiceBusSenderAsyncClient.MAX_MESSAGE_LENGTH_BYTES));
+        doNothing().when(onClientClose).run();
```

Comment:
by default the mock runnable does nothing.

Reply:
It was throwing NPE in test when it was invoked by client.